bridges/InternetArchiveBridge.php

```diff
@@ -0,0 +1,292 @@
+<?php
+class InternetArchiveBridge extends BridgeAbstract {
+	const NAME = 'Internet Archive Bridge';
+	const URI = 'https://archive.org';
+	const DESCRIPTION = 'Returns newest uploads, posts and more from an account';
+	const MAINTAINER = 'VerifiedJoseph';
+	const PARAMETERS = array(
+		'Account' => array(
+			'username' => array(
+				'name' => 'Username',
+				'type' => 'text',
+				'required' => true,
+				'exampleValue' => '@verifiedjoseph',
+			),
+			'content' => array(
+				'name' => 'Content',
+				'type' => 'list',
+				'values' => array(
+					'Uploads' => 'uploads',
+					'Posts' => 'posts',
+					'Reviews' => 'reviews',
+					'Collections' => 'collections',
+					'Web Archives' => 'web-archive',
+				),
+				'defaultValue' => 'uploads',
+			)
+		)
+	);
+
+	const CACHE_TIMEOUT = 900; // 15 mins
+
+	private $skipClasses = array(
+		'item-ia mobile-header hidden-tiles',
+		'item-ia account-ia'
+	);
+
+	public function collectData() {
+
+		$html = getSimpleHTMLDOM($this->getURI())
+			or returnServerError('Could not request: ' . $this->getURI());
+
+		if ($this->getInput('content') != 'posts') {
+
+			$detailsDivNumber = 0;
+
+			foreach ($html->find('div.results > div[data-id]') as $index => $result) {
+				$item = array();
+
+				if (in_array($result->class, $this->skipClasses)) {
+					continue;
+				}
+
+				if ($result->class === 'item-ia') {
+					if ($this->getInput('content') === 'reviews') {
+						$item = $this->processReview($result);
+					}
+
+					if ($this->getInput('content') === 'uploads') {
+						$item = $this->processUpload($result);
+					}
+				}
+
+				if ($result->class === 'item-ia url-item') {
+					$item = $this->processWebArchives($result);
+				}
+
+				if ($result->class === 'item-ia collection-ia') {
+					$item = $this->processCollection($result);
+				}
+
+				if ($this->getInput('content') != 'reviews') {
+					$hiddenDetails = $this->processHiddenDetails($html, $detailsDivNumber, $item);
+
+					$this->items[] = array_merge($item, $hiddenDetails);
+				} else {
+
+					$this->items[] = $item;
+
+				}
+
+				$detailsDivNumber++;
+			}
+		}
+
+		if ($this->getInput('content') === 'posts') {
+			$this->items = $this->processPosts($html);
+		}
+	}
+
+	public function getURI() {
+
+		if (!is_null($this->getInput('username')) && !is_null($this->getInput('content'))) {
+			return self::URI . '/details/' . $this->processUsername() . '&tab=' . $this->getInput('content');
+		}
+
+		return parent::getURI();
+	}
+
+	public function getName() {
+
+		if (!is_null($this->getInput('username')) && !is_null($this->getInput('content'))) {
+			$parameters = $this->getParameters();
+
+			$contentValues = array_flip($parameters['Account']['content']['values']);
```

Comment:
`$parameters` isn't necessary because we have direct access to the PARAMETERS array inside a class.

```suggestion
			$contentValues = array_flip(self::PARAMETERS['Account']['content']['values']);
```